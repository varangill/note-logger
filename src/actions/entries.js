import uuid from 'uuid';

// ADD_ENTRY
export const addEntry = (
  {
    description = '', //these are default values for if nothing is passed in
    note = '',
    tag = "task",
    createdAt = 0
  } = {}
) => ({ //addEntry returns this object
  type: 'ADD_ENTRY',
  entry: { //sets the entry equal to the variables from the inputs
    id: uuid(),
    description,
    note,
    tag,
    createdAt
  }
});

// REMOVE_ENTRY
export const removeEntry = ({ id } = {}) => ({
  type: 'REMOVE_ENTRY',
  id
});

// EDIT_ENTRY
export const editEntry = (id, updates) => ({
  type: 'EDIT_ENTRY',
  id,
  updates
});
