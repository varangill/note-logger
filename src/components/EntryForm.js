import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';


export default class EntryForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.entry ? props.entry.description : '',
      note: props.entry ? props.entry.note : '',
      tag: props.entry ? props.entry.tag : 'task',
      createdAt: props.entry ? moment(props.entry.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onTagChange = (e) => {
    const tag = e.target.value;
    this.setState(() => ({ tag }));
    console.log("tag")
  }
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.description) {
      this.setState(() => ({ error: 'Please provide a title.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        description: this.state.description,
        tag: this.state.tag,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <select 
            value={this.state.tag}
            onChange={this.onTagChange}
          >
            <option value="task">Task</option>
            <option value="goal">Goal</option>
          </select>
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Add a note for your entry (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          >
          </textarea>
          <button>Add Entry</button>
        </form>
      </div>
    )
  }
}
