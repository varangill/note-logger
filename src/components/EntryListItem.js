import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const EntryListItem = ({ id, description, tag, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {moment(createdAt).format('MMM Do, YYYY')}
      - 
      {tag}
    </p>
  </div>
);

export default EntryListItem;

// ATTEMPTS AT PIN FUNCTION
// export default class EntryListItem extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       description: props.entry ? props.entry.description : '',
//       note: props.entry ? props.entry.note : '',
//       alphabet: props.entry ? (props.entry.alphabet / 100).toString() : '',
//       tag: props.entry ? props.entry.tag : 'task',
//       createdAt: props.entry ? moment(props.entry.createdAt) : moment()
//     };
//   }
//   pinFunc = () => {
//     console.log("pinned!");
//   };
//   render () {
//     return (
//       <div>
//         <Link to={`/edit/${this.state.id}`}>
//           <h3>{this.state.description}</h3>
//         </Link>
//         <p>{this.state.alphabet} - {this.state.createdAt} - {this.state.tag}</p>
//       </div>
//     )
//   }
// }
