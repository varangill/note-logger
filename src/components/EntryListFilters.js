import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAlphabet, setStartDate, setEndDate, setTypeFilter } from '../actions/filters';

export class EntryListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'alphabet') {
      this.props.sortByAlphabet();
    }
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onTypeChange = (e) => {
    this.props.setTypeFilter(e.target.value);
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="alphabet">Alphabet</option>
        </select>
        
        <select //FIX THIS
          onChange={this.onTypeChange}
        >
          <option value="all">All</option>
          <option value="task">Task</option>
          <option value="goal">Goal</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  setTypeFilter: (type) => dispatch(setTypeFilter(type)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAlphabet: () => dispatch(sortByAlphabet()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(EntryListFilters);
