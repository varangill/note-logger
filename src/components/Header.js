import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {initLogout} from '../actions/auth';

export const Header = ({initLogout}) => (
  <header>
    <h1>NoteLogger</h1>
    <NavLink to="/main" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Entry</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    <button onClick={initLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  initLogout: () => dispatch(initLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
