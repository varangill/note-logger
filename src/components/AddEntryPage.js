import React from 'react';
import { connect } from 'react-redux';
import EntryForm from './EntryForm';
import { addEntry } from '../actions/entries';

export class AddEntryPage extends React.Component {
  onSubmit = (entry) => {
    this.props.addEntry(entry);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <h1>Add Entry</h1>
        <EntryForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addEntry: (entry) => dispatch(addEntry(entry))
});

export default connect(undefined, mapDispatchToProps)(AddEntryPage);
