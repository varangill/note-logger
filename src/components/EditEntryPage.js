import React from 'react';
import { connect } from 'react-redux';
import EntryForm from './EntryForm';
import { editEntry, removeEntry } from '../actions/entries';

export class EditEntryPage extends React.Component {
  

  onSubmit = (entry) => {
    this.props.editEntry(this.props.entry.id, entry);
    this.props.history.push('/');
  };

  onDelete = () => {
    this.props.removeEntry({ id: this.props.entry.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <EntryForm
          entry={this.props.entry}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onDelete}>Remove</button>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => {
  return {
    entry: state.entries.find((entry) => entry.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch, props) => ({
  editEntry: (id, entry) => dispatch(editEntry(id, entry)),
  removeEntry: (data) => dispatch(removeEntry(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditEntryPage);
