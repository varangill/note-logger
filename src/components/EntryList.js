import React from 'react';
import { connect } from 'react-redux';
import EntryListItem from './EntryListItem';
import selectEntries from '../selectors/entries';

export const EntryList = (props) => (
  <div>
    {
      props.entries.length === 0 ? (
        <p>No Entries</p>
      ) : (
        props.entries.map((entry) => {
          return <EntryListItem key={entry.id} {...entry} />; //removed return
        })
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    entries: selectEntries(state.entries, state.filters)
  };
};

export default connect(mapStateToProps)(EntryList);
