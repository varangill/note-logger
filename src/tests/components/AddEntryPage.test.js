import React from 'react';
import { shallow } from 'enzyme';
import { AddEntryPage } from '../../components/AddEntryPage';
import entries from '../fixtures/entries';

let addEntry, history, wrapper;

beforeEach(() => {
  addEntry = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddEntryPage addEntry={addEntry} history={history} />);
});

test('should render AddEntryPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('EntryForm').prop('onSubmit')(entries[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(addEntry).toHaveBeenLastCalledWith(entries[1]);
});
