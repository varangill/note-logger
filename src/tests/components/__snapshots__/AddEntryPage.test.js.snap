// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddEntryPage correctly 1`] = `
<div>
  <h1>
    Add Entry
  </h1>
  <EntryForm
    onSubmit={[Function]}
  />
</div>
`;
